src/App.tsx

```diff
@@ -1,39 +1,63 @@
 import React from 'react';
 import './App.scss';
+import { Clock } from './Clock';
+
+type State = {
+  clockName: string,
+  hasClock: boolean,
+};
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+export class App extends React.Component<{}, State> {
+  state: Readonly<State> = {
+    clockName: 'Clock-0',
+    hasClock: true,
+  };
+
+  intervalId = 0;
+
+  componentDidMount() {
+    document.addEventListener('contextmenu', this.removeClock);
+
+    document.addEventListener('click', this.addClock);
+
+    this.intervalId = window.setInterval(() => {
+      this.setState({ clockName: getRandomName() });
+    }, 3300);
+  }
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+  componentWillUnmount(): void {
+    document.removeEventListener('contextmenu', this.removeClock);
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+    document.removeEventListener('click', this.addClock);
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+    clearInterval(this.intervalId);
+  }
 
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
+  addClock = () => {
+    this.setState({ hasClock: true });
+  };
 
-        {' time is '}
+  removeClock = (event: any) => {
```

Comment:
It's not a good practice to use any for the event. You should try to find the correct type for it